packages/framework/src/Model/Mail/Mailer.php

```diff
@@ -5,20 +5,54 @@
 use Swift_Attachment;
 use Swift_Mailer;
 use Swift_Message;
+use Symfony\Component\HttpKernel\KernelInterface;
 
 class Mailer
 {
+    /**
+     * @var \Symfony\Component\HttpKernel\KernelInterface
+     */
+    protected $kernel;
+
     /**
      * @var \Swift_Mailer
      */
     protected $swiftMailer;
 
     /**
      * @param \Swift_Mailer $swiftMailer
+     * @param \Symfony\Component\HttpKernel\KernelInterface $kernel
      */
-    public function __construct(Swift_Mailer $swiftMailer)
+    public function __construct(Swift_Mailer $swiftMailer, KernelInterface $kernel)
     {
         $this->swiftMailer = $swiftMailer;
+        $this->kernel = $kernel;
+    }
+
+    public function flushSpoolQueue()
+    {
+        $container = $this->kernel->getContainer();
+        if (!$container->has('mailer')) {
+            return;
+        }
+        $mailers = array_keys($container->getParameter('swiftmailer.mailers'));
+        foreach ($mailers as $name) {
+            if (method_exists($container, 'initialized') ? $container->initialized(sprintf('swiftmailer.mailer.%s', $name)) : true) {
+                if ($container->getParameter(sprintf('swiftmailer.mailer.%s.spool.enabled', $name))) {
+                    $mailer = $container->get(sprintf('swiftmailer.mailer.%s', $name));
+                    $transport = $mailer->getTransport();
+                    if ($transport instanceof \Swift_Transport_SpoolTransport) {
```

Comment:
Is there a way that this won't be SpoolTransport? Because you have a condition that spool needs to be enabled for the mailer and maybe this line is not necessary. I don't know the whole logic behind it it would just make sense to me.

Reply:
hold-on EmailSenderListener.php::onTerminate